File: node/send.js
import { Sendpository, SendpositoryError } from "sendpository";

const to = process.argv[2];
if (!to) {
  console.error("Usage: npm run send -- you@example.com");
  process.exit(1);
}

const sendpository = new Sendpository(process.env.SENDPOSITORY_API_KEY, {
  // Only set when pointing at a non-default API, e.g. a self-hosted install.
  baseUrl: process.env.SENDPOSITORY_BASE_URL || undefined,
});

try {
  const { id } = await sendpository.emails.send({
    from: process.env.SENDPOSITORY_FROM,
    to: [to],
    subject: "Hello from Sendpository",
    html: "<p>It works. This email was sent with <strong>one API call</strong>.</p>",
    text: "It works. This email was sent with one API call.",
  });
  console.log(`Sent. Message id: ${id}`);
} catch (err) {
  if (!(err instanceof SendpositoryError)) throw err;
  console.error(`${err.type}: ${err.message}`);
  process.exit(1);
}

File: python/send.py
"""Send one email with the Sendpository API. Standard library only."""

import json
import os
import sys
import urllib.error
import urllib.request


def load_env(path=".env"):
    """Read KEY=value lines into os.environ, without overriding real env vars."""
    if not os.path.exists(path):
        return
    with open(path) as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith("#") and "=" in line:
                key, value = line.split("=", 1)
                os.environ.setdefault(key.strip(), value.strip().strip('"'))


def main():
    load_env()
    if len(sys.argv) < 2:
        sys.exit("Usage: python send.py you@example.com")

    base = os.environ.get("SENDPOSITORY_BASE_URL") or "https://api.sendpository.com/v1"
    request = urllib.request.Request(
        f"{base}/emails",
        method="POST",
        headers={
            "Authorization": f"Bearer {os.environ['SENDPOSITORY_API_KEY']}",
            "Content-Type": "application/json",
        },
        data=json.dumps(
            {
                "from": os.environ["SENDPOSITORY_FROM"],
                "to": [sys.argv[1]],
                "subject": "Hello from Sendpository",
                "html": "<p>It works. This email was sent with <strong>one API call</strong>.</p>",
                "text": "It works. This email was sent with one API call.",
            }
        ).encode(),
    )

    try:
        with urllib.request.urlopen(request, timeout=10) as res:
            print(f"Sent. Message id: {json.load(res)['id']}")
    except urllib.error.HTTPError as e:
        error = json.load(e)["error"]
        sys.exit(f"{error['type']}: {error['message']}")


if __name__ == "__main__":
    main()
